airlock/file_browser_api.py

```diff
@@ -371,21 +421,9 @@ def build_path_tree(
                 request_filetype=container.request_filetype(path),
             )
 
-            # If it has decendants, it is a directory. However, an empty
-            # directory in workspace still needs to be classed as a PathType.DIR.
-            # So we infer if it is by checking for lack of suffix. All output files
-            # *must* have a suffix, so this is a reasonable check.
-            #
-            # However, in theory, there could be an empty directory with
-            # a suffix in its name, so this will treat these as a file rather
-            # than a directory.  If this is a problem, we could instead call
-            # is_dir(), but we are trying to avoid hitting the filesystem in
-            # the tree recursion for speed.
-            #
-            # We have a backstop check to not blow up in this case
-            # PathItem.contents()
-            if descendants or path.suffix == "":
+            if descendants or (directories and path in directories):
```

Comment:
I guess it doesn't matter if `directories` is all directory paths or only empty ones, because the descendants check here will get the non-empty ones.  But if `scantree` added all dir paths instead of only empty ones, could we skip the descendants check here?

Reply:
We could be it would make pathlist much larger, as it would include all the directories.

For example, in my local 45k node tree, as currently is, `len(pathlist) == 45000`. If I include directories, its 71k+.  And we don't need explicit parent directories - that information is parsed out of the paths.  So I opted just to track files, and the occasionally extra empty dir.

This aligns with the API for requests data and git repo data two, where we only have a flat list of files.  We would have to manually reconstruct pathlist entries for those if we had every directory in the pathlist.